packages/sdk/src/cross-chain-messenger.ts

```diff
@@ -2301,15 +2301,47 @@ export class CrossChainMessenger {
       }
 
       if (this.bedrock) {
-        const withdrawal = await this.toLowLevelMessage(resolved, messageIndex)
-        return this.contracts.l1.OptimismPortal.populateTransaction.finalizeWithdrawalTransaction(
+        // get everything we need to finalize
+        const [status, withdrawal] = await Promise.allSettled([
+          this.getMessageStatus(message),
```

Comment:
Using `getMessageStatus` here works but runs the risk of doing a bunch of RPC calls in the happy path since the query required to check for `READY_FOR_RELAY` is expensive. Simply checking if the message is inside of `failedMessages` would guarantee a single RPC call and would avoid needing to add `READY_FOR_REPLAY` in this PR.

Reply:
true this is what I landed on in viem too. I was gonna use getWithdrawalStatus but ended up just checking failedMessage